packages/sdk/src/chains/BaseAssetTransfer.ts

```diff
@@ -43,4 +58,71 @@ export abstract class BaseAssetTransfer {
 
     return transfer;
   }
+
+  /**
+   * @param {Transfer} transfer Transfer to check
+   * @param {String} destinationProviderUrl URL of the destination chain provider
+   * @returns {Promise<Boolean>} Flag indicating whether there is sufficient balance to execute the transfer on the destination chain
+   */
+  async checkDestinationChainBalance(
+    transfer: Transfer<TransferType>,
+    destinationProviderUrl: string,
+  ): Promise<boolean> {
+    const destinationDomain = this.config.getDomainConfig(transfer.to.id);
+    const handlerAddress = destinationDomain.handlers.find(
+      h => h.type === ResourceType.FUNGIBLE,
+    )?.address;
+
+    if (!handlerAddress) {
+      throw new Error('No Funglible handler configured on destination domain');
+    }
+
+    const destinationResource = destinationDomain.resources.find(
+      r => r.resourceId === transfer.resource.resourceId,
+    );
+
+    if (destinationResource?.burnable) {
+      return true;
+    }
+
+    let handlerBalance;
+    let providerOrApiPromise;
+
+    switch (destinationDomain.type) {
```

Comment:
I think also this switch statement can be it's own function, in order to avoid having a big utility function

Reply:
I think the functions should be in the case per network.